Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -433,20 +433,40 @@ public override void Update(float frameTime)
 
                 if (flammable.FireStacks > 0)
                 {
-                    var air = _atmosphereSystem.GetContainingMixture(uid);
+                    var air = _atmosphereSystem.GetContainingMixture(uid, false, true);
 
-                    // If we're in an oxygenless environment, put the fire out.
-                    if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                    if (flammable.RequireOxygen)
                     {
-                        Extinguish(uid, flammable);
-                        continue;
+                        // If we're in an oxygenless environment, put the fire out.
+                        if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                        {
+                            Extinguish(uid, flammable);
+                            continue;
+                        }
                     }
 
                     var source = EnsureComp<IgnitionSourceComponent>(uid);
                     _ignitionSourceSystem.SetIgnited((uid, source));
 
                     if (TryComp(uid, out TemperatureComponent? temp))
-                        _temperatureSystem.ChangeHeat(uid, 12500 * flammable.FireStacks, false, temp);
+                    {
+                        // get hot
+                        var maxDeltaT = flammable.PeakFlameTemperature - temp.CurrentTemperature;
+                        if (maxDeltaT > 0) // we dont want to burn and get colder.
+                        {
+                            var tempBump = MathF.Sqrt(maxDeltaT) * flammable.FireStacks * 25; //monotonic function, slows as it reaches maxT
+                            _temperatureSystem.ChangeHeat(uid, tempBump, false, temp);
+                        } 
+
+                        // release gas
+                        if (air != null && flammable.EmissiveGasMix != null)
+                        {
+                            flammable.EmissiveGasMix.Temperature = temp.CurrentTemperature;
+                            var releasingGas = new GasMixture(flammable.EmissiveGasMix.Volume);
+                            _atmosphereSystem.Merge(releasingGas, flammable.EmissiveGasMix); // generate gas for the entity's entire burning lifespan, but we use ReleaseGasTo to handle temperature/pressure for us since those will change for other reasons.
+                            _atmosphereSystem.ReleaseGasTo(releasingGas, air, releasingGas.Pressure); // doing it this way means that we dont need to track the entity damage values or know its destructible or mobstate thresholds.
+                        }                                                                    // the downside, and a possible later rework, is that you ""lose"" some gas if the candle is burning in an atmosphere where it can't emit due to pressure.
```

Comment:
So why not `Merge` into the target?

Reply:
Merge doesn't respect the pressure differential, and this way is slowly slow down how much gas is emitting when the pressures are similar.

Reply:
Why should it respect a pressure differential at all? You have something contributing gas -- it should spread out instantly since ideal gas law.

Reply:
We're considering phase change here, so pressure (and temperature) should play a part in determining the state. It might be the case I just need to bite the bullet and handle the destruct thresholds so the entity extinguishes if it cant emit.

I thought this was a more interesting gameplay mechanic however because it means you cant just light the candle and forget about it, allowing the pressure to hold the candle together -- you need to handle the pressure or ""lose"" the gas.

Reply:
What is changing phase? This is a decomposition of a solid, not a phase change. The "proper" solution would be to require activation energy for the reaction and simulate reaction rates. I think this would be overkill.

> I thought this was a more interesting gameplay mechanic however because it means you cant just light the candle and forget about it

I don't think this makes for interesting gameplay.